crates/primitives/src/transaction/mod.rs

```diff
@@ -986,6 +1023,41 @@ impl From<TransactionSigned> for TransactionSignedNoHash {
     }
 }
 
+/// A [`TransactionSignedNoHash`] wrapper used to performs proptest tests on it,
+/// while limiting alloy TxLegacy gas_limit to u64::MAX in order to match TxLegacy Compact type.
+#[cfg_attr(any(test, feature = "reth-codec"), reth_codecs::add_arbitrary_tests(compact))]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+struct MockTransactionSignedNoHash(TransactionSignedNoHash);
+
+#[cfg(any(test, feature = "arbitrary"))]
+impl<'a> arbitrary::Arbitrary<'a> for MockTransactionSignedNoHash {
```

Comment:
this is only used for testing, right?

could we move this into the test module?